### PR TITLE
Chore: updating User Profile with null model or null avatar 

### LINKED ISSUE #4755: Review consequences of inability to load avatars
**Abstract**
Null (not initialized) fields cause troubles in random places in the code that are very hard to track and fight.
Those can be not only failures but significant delays due to asynchronous nature of loading.

**Acceptance Criteria**

Review 2 following cases:
1. `UserProfile.avatar` is not loaded [yet] but used in views and other places in the code
2. Avatar Load Failed for other users, how it affects interaction with them

Review how repetitions are done and whether recovering gracefully from a not-loaded avatar is possible.
As a result, discuss and establish rules that will make the overall behavior clean and clear if avatars are not loaded.

**Propositions:**
1. Manually simulate loading failure (hardcode) and observe the consequences

### PR DESCRIPTION
**Pull request by popuz**
## What does this PR change?
Closes #4755

- add fallback or recovering UserProfile in the case it was failing from payload parsing

## How to test the changes?

Since it was an edge-case, I am not sure it can be tested by QA. 
However, I expect it to fix issue #5058 (if it still present). In case it is reproducible, you should see a guy in just black pants with normal skin color (instead of black in world and transparent in AvatarEditor)

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at 9724acc</samp>

This pull request enhances the `UserProfile` feature by using new C# features, adding error handling, and removing unused code. It also updates the `ProfileHUDViewV2` and `AvatarModel` classes to use new C# syntax for readability and consistency. Finally, it adds a new assembly reference to the `UserProfile.asmdef` file to enable feature flagging.

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ProfileHUD/ProfileHUDViewV2.cs [hunk 1]
```diff
@@ -304,9 +304,9 @@ private void HandleUnverifiedProfileName(UserProfile userProfile)
     {
         textName.text = string.IsNullOrEmpty(userProfile.userName) || userProfile.userName.Length <= NAME_POSTFIX_LENGTH
             ? userProfile.userName
-            : userProfile.userName.Substring(0, userProfile.userName.Length - NAME_POSTFIX_LENGTH - 1);
+            : userProfile.userName[..(userProfile.userName.Length - NAME_POSTFIX_LENGTH - 1)];
 
-        textPostfix.text = $"#{userProfile.userId.Substring(userProfile.userId.Length - NAME_POSTFIX_LENGTH)}";
+        textPostfix.text = $"#{userProfile.userId[^NAME_POSTFIX_LENGTH..]}";
```
Comment (Kinerius) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ProfileHUD/ProfileHUDViewV2.cs:309]: TIL

**Review by Kinerius — APPROVED** [review]

File: unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfile.cs [hunk 2]
```diff
@@ -39,27 +41,76 @@ public enum EmoteSource
...
     private int emoteLamportTimestamp = 1;
 
     public void UpdateData(UserProfileModel newModel)
     {
         if (newModel == null)
         {
-            model = new UserProfileModel();
-            return;
+            if (DataStore.i.featureFlags.flags.Get().IsFeatureEnabled("user_profile_null_model_exception"))
+                Debug.LogError("Model is null when updating UserProfile! Using fallback or previous model instead.");
```
Comment (dalkia) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfile.cs:99]: Just to confirm, this sends a Sentry error?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfile.cs [hunk 3]
```diff
@@ -79,14 +130,13 @@ public void UpdateData(UserProfileModel newModel)
...
+        if (faceSnapshotDirty)
             snapshotObserver.RefreshWithUri(face256SnapshotURL);
 
-        if (model.snapshots != null && bodySnapshotDirty)
+        if (bodySnapshotDirty)
             bodySnapshotObserver.RefreshWithUri(bodySnapshotURL);
 
-        if (isModelDirty)
-            OnUpdate?.Invoke(this);
+        OnUpdate?.Invoke(this);
```
Comment (dalkia) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfile.cs:139]: To confirm, is it safe to remove the `isModelDirty` bool? Wouldnt then the `OnUpdate` function be called even if the model has no changes?

**Review by dalkia — APPROVED** [review]
Approving, although I have two question to confirm

File: unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfile.cs (hunk 2 above)
Comment (popuz) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfile.cs:99]: I confirmed it with Ashley and it should work (regarding [this Sentry documentation page for Unity](https://docs.sentry.io/platforms/unity/)). 
```

public class TestMonoBehaviour : MonoBehaviour
{
    private GameObject testObject = null;

    void Start()
    {
        Debug.Log("Captured Log");              // Breadcrumb
        Debug.LogWarning("Captured Warning");   // Breadcrumb
        Debug.LogError("Captured Error");       // Captured Error

        // This will throw an unhandled Null Reference Exception
        testObject.GetComponent<Transform>();   // Captured error
    }
}
```

File: unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfile.cs (hunk 3 above)
Comment (popuz) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfile.cs:139]: I think that `isModelDirty` was a small optimization. However, many clients of this class are highly dependent on this callback. Thus, not calling it can cause some systems wait it for long (possibly infinite) time. For example in the bug-report #5058 you can see QA noticed very long loading time. It was exactly because LoadingScreen or other systems were awaiting OnUpdate to happen

**Review by kwssbocian — APPROVED** [review]
✔️No occurrences of transparent or otherwise incorrect looking avatars. No other PR-specific issues observed.
